src/products/components/ProductVariants/ProductVariants.tsx

```diff
@@ -233,6 +284,50 @@ export const ProductVariants: React.FC<ProductVariantsProps> = props => {
                       )}
                     </TableCell>
                   </Hidden>
+                  <TableCell
```

Comment:
Can we move this logic out of template?
```
const availabilityMessage = (warehouse, numAvailable, variant) => {
    if (warehouse === null && numAvailable === 0) {
        return {
            defaultMessage="Unavailable in all locations"
            description="product variant inventory"
    }
    if (warehouse !=== null && numAvailable > 0) {
        return {
            defaultMessage="{numAvailable} available at {numLocations} {numLocations,plural,one{location} other{locations}}"
            description="product variant inventory"
            values={{
            numAvailable,
            numLocations: variant.stock.length
            }
        }
    }    
    return {                          
        defaultMessage="Unavailable"
        description="product variant inventory"}
    }
    
}

render (
  ...
    {numAvailable === null ? (
        <Skeleton />
    ) : (
        <FormattedMessage ...availabilityMessage  />
    )
)

```

Reply:
I guess we could, but in a slightly different way (we need to use `intl.formatMessage()` or `<FormattedMessage />` literals).